tck/src/main/java/org/eclipse/microprofile/lra/tck/service/spi/LRARecoveryService.java

```diff
@@ -0,0 +1,80 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck.service.spi;
+
+import java.net.URI;
+
+/**
+ * This SPI is providing the implementation with the ability to define what actions need to be taken
+ * in case the TCK needs to wait for the the delivery of of participant callbacks (Complete, Compensate, ...) when the
+ * LRA is asked to perform some eventually consistent action {@link LRARecoveryService#waitForCallbacks(URI)} and to
+ * define what actions can be taken to make the recovery (replay of the Compensate, Complete or Status) calls happen
+ * when the TCK requires to perform such actions {@link LRARecoveryService#waitForRecovery(URI)}.
+ */
+public interface LRARecoveryService {
```

Comment:
I read this doc as LRARecoveryService being passive (except for the examples). It just wait until the callbacks are done.

But If I remember correct, the the idea of the LRARecoveryService was that it actively forced the callbacks to be performed faster (by poking some internal methods so that callbacks are done)  (at least for the waitForRecovery)

The waitForCallback is indeed just 'waiting' for a short period so that implementation can deliver the callbacks and they are executed before the test method verifies if those callbacks are executed.

Reply:
This SPI is also allowed to throw Exceptions in case it detects that waiting or forcing recovery is not possible.

That way the TCK test also fails and covers for the case that it should wait indefinitely for a callback to happen which is not going to be performed.

Reply:
good points. I will add this.